services/com.mde-network.ep.toolfunctions.xtextfunction/src/main/java/com/mdenetnetwork/ep/toolfunctions/xtextfunction/RunXtextFunction.java

```diff
@@ -20,12 +20,18 @@ public void serviceImpl(JsonObject request, JsonObject response) throws Exceptio
 		
 		ByteArrayOutputStream bos = new ByteArrayOutputStream();
 		
+        String validator = request.get("validator") != null ? request.get("validator").getAsString() : null;
+        String scopeprovider = request.get("scopeprovider") != null ? request.get("scopeprovider").getAsString() : null;
+        String generator = request.get("generator") != null ? request.get("generator").getAsString() : null;
+
 		new XtextTool().
 			run( request.get("languageName").getAsString() , 
 				 request.get("baseName").getAsString(), 
 				 request.get("extension").getAsString(),
 				 request.get("grammar").getAsString(),
-				 parseProjectFiles(request.get("projectFiles").toString()),
+				 validator,
+				 scopeprovider,
+				 generator,
```

Comment:
We can now also delete function `parseProjectFiles`, right?

Reply:
Agree, removed in 766e28f1cad63b46f7f0ed02028e89688b1e94c4